stl/inc/xstring

```diff
@@ -2471,7 +2471,7 @@ private:
     static constexpr size_t _Memcpy_val_size   = sizeof(_Scary_val) - _Memcpy_val_offset;
 
     template <class _Iter>
-    using _Is_elem_cptr = bool_constant<_Is_any_of_v<_Iter, const _Elem* const, _Elem* const, const _Elem*, _Elem*>>;
+    static constexpr bool _Is_elem_cptr = bool_constant<_Is_any_of_v<_Iter, const _Elem* const, _Elem* const, const _Elem*, _Elem*>>;;

```

Comment:
Note that [`bool_constant` is an alias template](https://en.cppreference.com/w/cpp/types/integral_constant#Helper_alias_templates) which can't be used as an initializer. I think what we need is its argument which is a `bool` constant value.
And it's unnecessary to repeat `;`.

Reply:
> Note that [`bool_constant` is an alias template](https://en.cppreference.com/w/cpp/types/integral_constant#Helper_alias_templates) which can't be used as an initializer. I think what we need is its argument which is a `bool` constant value.

Ok, i should I fix it?

Reply:
Yeah, `bool_constant` was needed when `_Is_elem_cptr` was defined as an alias template. Now `_Is_elem_cptr` is a variable template, it should be removed.

Reply:
>And it's unnecessary to repeat `;`.
I just fixed it.

Reply:
> Yeah, `bool_constant` was needed when `_Is_elem_cptr` was defined as an alias template. Now `_Is_elem_cptr` is a variable template, `bool_constant` should be removed.

So it needs to look something like this? :
`static constexpr bool _Is_elem_cptr =_Is_any_of_v<_Iter, const _Elem* const, _Elem* const, const _Elem*, _Elem*>;`

Reply:
Right, but remember to run `clang-format` to get the file correctly formatted.

Reply:
> So it needs to look something like this? :
> `static constexpr bool _Is_elem_cptr =_Is_any_of_v<_Iter, const _Elem* const, _Elem* const, const _Elem*, _Elem*>;`

Yep. Just go ahead.

Reply:
> Yep. Just go ahead.

Done.